### ISSUE TITLE
Bounding Box Cutting Off Object

### ISSUE THREAD

**Issue by daylanc** [AlexeyAB/darknet#2072]
I did a quick training on the number 0 and 2. I used LabelImg to created the annotation .txt files. After training my dataset of about 800 images (300-0's, 300-2's, 200-other) I tested it with images and noticed that the resulting bounding box cuts off part of the numbers. In the image below, all detected 0's and 2's are cut off. It still recognizes the numbers correctly, but doesn't look like the "dog, bike, bus" example on the yolo website. Result image:

![predictions](https://user-images.githubusercontent.com/34255535/50198874-c3fca880-02f1-11e9-8d30-3a37e3c665bb.jpg)

**Comment by AlexeyAB** [comment]
* How many iterations did you train?

* Label bounded boxes of objects should me more in your training dataset.

**Comment by daylanc** [comment]
@kooscode
I’m using v3 any my network size is the default from pjreddie (608x608). All the images I trained with were 512x720. Should I change my network to 512x720? I did not change the anchor boxes. I was reading some issues on that, but I’m having a hard time figuring out what that is and how to calculate them.

**Comment by daylanc** [comment]
@AlexeyAB
I trained my network 2 separate times. The first time was with 300 images of a single class (#2 only) and my second time was the same 300 images plus 500 more (300-0’s, 200-other) for a total of 800. I stopped the first training at 10,000 iterations and the second at 8,000 (I used the pretrained model results from the first training run). I split my training and valid sets at 80% train and 20% valid for both.

**Comment by AlexeyAB** [comment]
@daylanc Can you show your training images with drawed labels?

**Comment by kooscode** [comment]
@daylanc why not resize the network to 416x416? it would train and inference a lot faster, also like AlexeyAB asked, lets look at the labeled data.

for anchor boxes you need to to try and get them to match the shape of your typical detections, for example if you are going to detect a mostly peope, pedeistrians and road signs, the bbox shapes (mostly large rectangles) will look very different than if you are only going to detect plam trees from arial images.. (small squares), i;ve seen misalignment in anchor box mismatches and also if your network input size is not an "odd" division of 32.. so the region proposals does not have a row right in the center of the image and if most of your subjects are expected to be image center, they will miss-align.. 

also, try and train with images larger than your input size so the bounding boxes are clrearly distinguished from the background. 

my 2 cents.

**Comment by daylanc** [comment]
@AlexeyAB 
My dataset had these images in it:

![screenshot from 2018-12-12 09-27-02](https://user-images.githubusercontent.com/34255535/50266794-21aff400-03c9-11e9-8d09-238601fbf8db.png)

![screenshot from 2018-12-12 10-31-27](https://user-images.githubusercontent.com/34255535/50266827-4015ef80-03c9-11e9-9f53-eac0c76e85ce.png)


![screenshot from 2018-12-16 22-14-56](https://user-images.githubusercontent.com/34255535/50266815-33919700-03c9-11e9-8179-d91794c62dd3.png)

![screenshot from 2018-12-16 22-15-58](https://user-images.githubusercontent.com/34255535/50266835-460bd080-03c9-11e9-9f1f-0b960dcf628d.png)

Like I said, I tried to keep it very very very simple just to start. I was planning on getting more complex once I got the correct process down.

**Comment by daylanc** [comment]
@kooscode 
I will try resizing the network this weekend and see if the results change. While creating the dataset, I wrote the numbers in various sizes (refer to previous comment to Alexey). When running the detector, I also adjust the test images to 512x720. I could adjust it to like 480x512 and see if that produces better results??

**Comment by AlexeyAB** [comment]
@daylanc 

* Try to increase size of bounded boxes in your Training dataset


----

If it doesn't help - also:

* you can try to change `ignore_thresh = .7` to `ignore_thresh = .8`

* you can try to add this line `scale = scale * 4;` here: https://github.com/AlexeyAB/darknet/blob/49189c22e8c09ba7c11622ec22a72b3678887ed2/src/yolo_layer.c#L98

**Comment by daylanc** [comment]
@AlexeyAB
When you say to increase the size of the bounding boxes in the training dataset do you mean to leave more white space around the numbers?

**Comment by AlexeyAB** [comment]
@daylanc Yes.

**Comment by daylanc** [comment]
@AlexeyAB 
Okay I’ll give that a try this weekend and let you know the results

**Comment by daylanc** [comment]
> 
> If it doesn't help - also:
> 
>     * you can try to change `ignore_thresh = .7` to `ignore_thresh = .8`
> 
>     * you can try to add this line `scale = scale * 4;` here: 
>         
>           
>             [darknet/src/yolo_layer.c](https://github.com/AlexeyAB/darknet/blob/49189c22e8c09ba7c11622ec22a72b3678887ed2/src/yolo_layer.c#L98)
>           
>           
>                Line 98
>             in
>             [49189c2](/AlexeyAB/darknet/commit/49189c22e8c09ba7c11622ec22a72b3678887ed2)

@AlexeyAB 
What do you mean by `ignore_thresh = .8`? I cannot find `ignore_thresh` variable in `darknet/src/yolo_layer.c`.
